modules/vmss/main.tf

```diff
@@ -81,6 +89,7 @@ resource "azurerm_linux_virtual_machine_scale_set" "this" {
       ip_configuration {
         name                                         = "primary"
         primary                                      = true
+        version                                      = "IPv4"
```

Comment:
What do you think about NOT hardcoding IPv4, but make it flexible (e.g. by using optionals) ?
According to current documentation, *`version` is the Internet Protocol Version which should be used for IP Configuration. Possible values are IPv4 and IPv6. Defaults to IPv4*.

Reply:
At the moment we don't support IPv6 in the modules at all. That's what we would do if we were to introduce it but it would be a much bigger change in multiple places in the code. Not in scope of this PR.

Reply:
Still in my opinion we should not hardcode it now, if we know, that in future IPv6 may be added (I'm not suggesting to add full IPv6 support in this PR). Why not define this as an attribute, which can be optional with default value `IPv4` and with validation that only `IPv4` is supported ? Then in future it can be easily extended for `IPv6`.

While doing global refactor of all modules it was suggested and preferred approach e.g. we have it for `protocol` in `gwlb` module: 

https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/blob/1523e64f2576a8c578d15ccbb80523db9de929f8/modules/gwlb/variables.tf#L181

where we defined validation for only 1 possible option:

https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/blob/1523e64f2576a8c578d15ccbb80523db9de929f8/modules/gwlb/variables.tf#L191-L198

Reply:
I just added this to remember about it when we introduce IPv6 across the modules which is definitely in the roadmap, so we follow other public clouds. I don't want to introduce any IP protocol version variables yet but once we do it, we'll need to do so in almost every module. Let's keep it out of this PR.

Reply:
If `IPv4` is default option, then what is the reason to hard code it in the module source code (instead just leaving it as it was before with default version define in AzureRM provider) ?

Reply:
> I just added this to remember about it when we introduce IPv6 across the modules

Reply:
I'm not a supporter of this approach, but I don't want to block you. 
PR is approved.